docker-compose.dev.yml

```diff
@@ -0,0 +1,51 @@
+version: '3.7'
+
+services:
+  app:
+    image: mysocietyorg/caps-dev:${TAG:-latest}
+    build:
+      context: .
+      dockerfile: Dockerfile.dev
+    command: /usr/src/app/script/server
+    volumes:
+      - solr_data:/opt/solr/server/solr/mycores
+      - ./:/usr/src/app/
+      - ./conf/config.py.docker:/usr/src/app/conf/config.py
+    environment:
+      - REPOSITORY_DB_PASS=${REPOSITORY_DB_PASS:-caps}
+      - SECRET_KEY=${SECRET_KEY:-xx}
+      - DEBUG=${DEBUG:-1}
+      - MAPIT_API_KEY=${MAPIT_API_KEY:-xxx}
+      - GOOGLE_ANALYTICS=${GOOGLE_ANALYTICS:-xxx}
+      - MAILCHIMP_KEY=${MAILCHIMP_KEY:-xxx}
+    depends_on:
+      - postgres
+      - solr
+    ports:
+      - ${PORT:-8000}:8000
+
+  postgres:
+    image: postgres:11
+    volumes:
+      - postgres_data:/var/lib/postgresql/data/
+    environment:
+      - POSTGRES_USER=${REPOSITORY_DB_USER:-caps}
+      - POSTGRES_PASSWORD=${REPOSITORY_DB_PASS:-caps}
+      - POSTGRES_DB=${REPOSITORY_DB_NAME:-caps}
+
+  solr:
+    image: solr:6
+    volumes:
+      - solr_data:/opt/solr/server/solr/mycores
+      - ./conf/schema.xml:/opt/solr/server/solr/configsets/basic_configs/conf/schema.xml
+      - ./conf/solrconfig.xml:/opt/solr/server/solr/configsets/basic_configs/conf/solrconfig.xml
+    command:
+      - solr-precreate
+      - caps
+      - /opt/solr/server/solr/configsets/basic_configs
+    ports:
+      - 8983:8983
+      
+volumes:
+  postgres_data:
+  solr_data:
```

Comment:
I am not sure if you can avoid redeclaring the postgres and solr services altogether, it's not totally clear from skimming the docs.

Reply:
Annoyingly you can't extend a service with a 'depends_on', and the canonical solution to this would be to move our 'app' into a separate base .yml, and reimport and extend it in both places. That seems messier in terms of the clarity of the normal docker-compose file. Will do the extends on the other services though.

Reply:
oh. how annoying.